### ISSUE TITLE
Add Option: "Git: Ignore new commits in submodules"

### ISSUE THREAD

**Issue by Fmstrat** [microsoft/vscode#78016]
In some projects, monorepos with submodules are kept up to date with the latest commits. In others, they are centralized storage and are often behind the submodules.

In the later case, VSCode ends up highlighting every submodule as a change. For instance, take this case:

```
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)
  (commit or discard the untracked or modified content in submodules)

        modified:   submodules/app1 (modified content)
        modified:   submodules/app2 (new commits)
        modified:   submodules/app3 (new commits)
        modified:   submodules/app4 (new commits)
        modified:   submodules/app5 (new commits)
        modified:   submodules/app6 (new commits)

no changes added to commit (use "git add" and/or "git commit -a")
```
Right now, `app1` through `app6` are highlighted as changed. I'd like to request an option called `Git: Ignore new commits in submodules` that would remove the highlighting from `app2` through `app6`.

**Comment by joaomoreno** [comment]
But... if `git status` doesn't ignore those changes (as shown in your console output)... why would VS Code?

**Comment by Fmstrat** [comment]
> But... if `git status` doesn't ignore those changes (as shown in your console output)... why would VS Code?

Thanks for the response @joaomoreno 

`git` offers many ways to filter the content. For instance `git diff` has a `submodule` option, or a `regex filter (-G)` option, both of which are used to filter outputs via aliases for exactly this type of thing.

I took a look at the vscode source (wow it's clean) and see how you pass in args to `diffFiles` so I've forked and am putting together a PR in my spare time that you guys could take a look at. 

Thanks!
